libraries/Microsoft.Bot.StreamingExtensions/ConcurrentStream.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.StreamingExtensions.Payloads;
+
+namespace Microsoft.Bot.StreamingExtensions
+{
+    internal class ConcurrentStream : Stream
+    {
+        private readonly ContentStreamAssembler _assembler;
+        private readonly Queue<byte[]> _bufferQueue = new Queue<byte[]>();
+
+        private readonly SemaphoreSlim dataAvailable = new SemaphoreSlim(0, int.MaxValue);
+        private readonly object syncLock = new object();
```

Comment:
This should be SemaphoreSlim(0,1). Not sure why you're using int.MaxValue. You are using this as a boolean that you can wait on (which is legit), but for that you only need 0,1.